controllers/tasksRequests.js

```diff
@@ -43,6 +46,73 @@ const fetchTaskRequestById = async (req, res) => {
   }
 };
 
+const addTaskRequests = async (req, res) => {
+  try {
+    const taskRequestData = req.body;
+    const usernamePromise = usersUtils.getUsername(taskRequestData.userId);
+    if (req.userData.id !== taskRequestData.userId && !req.userData.roles?.super_user) {
+      return res.boom.forbidden("Not authorized to create the request");
+    }
+    if (taskRequestData.proposedDeadline < taskRequestData.proposedStartDate) {
+      return res.boom.badRequest("Task deadline cannot be before the start date");
+    }
+    switch (taskRequestData.requestType) {
+      case TASK_REQUEST_TYPE.ASSIGNMENT: {
+        const taskDataPromise = tasksModel.fetchTask(taskRequestData.taskId);
+
+        const [{ taskData }, username] = await Promise.all([taskDataPromise, usernamePromise]);
+        taskRequestData.taskTitle = taskData?.title;
+        if (!username) {
+          return res.boom.badRequest("User not found");
+        }
+        if (!taskData) {
+          return res.boom.badRequest("Task does not exist");
+        }
+        break;
+      }
+      case TASK_REQUEST_TYPE.CREATION: {
+        let issuePromise;
+        try {
+          const url = new URL(taskRequestData.externalIssueUrl);
+          const issueUrlPaths = url.pathname.split("/");
+          const repositoryName = issueUrlPaths[3];
+          const issueNumber = issueUrlPaths[5];
+          issuePromise = githubService.fetchIssuesById(repositoryName, issueNumber);
+        } catch (error) {
+          return res.boom.badRequest("External issue url is not valid");
+        }
+        const [issueData, username] = await Promise.all([issuePromise, usernamePromise]);
+        taskRequestData.taskTitle = issueData?.title;
+        if (!username) {
+          return res.boom.badRequest("User not found");
+        }
+        if (!issueData) {
+          return res.boom.badRequest("Issue does not exist");
+        }
+        break;
+      }
+    }
+    const newTaskRequest = await taskRequestsModel.createRequest(taskRequestData, req.userData.username);
+    if (newTaskRequest.isCreationRequestApproved) {
+      return res.boom.conflict("Task exists for the given issue.");
+    }
+    if (newTaskRequest.alreadyRequesting) {
+      return res.boom.badRequest("Task was already requested");
+    }
```

Comment:
- should we combine `isCreationRequestApproved` & `alreadyRequesting` into a single variable `status` where status can be an enum of `REQUESTED`, `APPROVED`, and any other possible states?